lib/libzpool/kernel.c

```diff
@@ -813,11 +813,9 @@ random_get_pseudo_bytes(uint8_t *ptr, size_t len)
 int
 ddi_strtoull(const char *str, char **nptr, int base, u_longlong_t *result)
 {
-	(void) nptr;
-	char *end;
-
+	char* end = NULL;
 	*result = strtoull(str, &end, base);
-	if (*result == 0)
```

Comment:
It's not necessarily an error for `strtoull` to return zero, if it actually parsed a zero from the string (e.g. `0 1 2 3`.

Reply:
hah yep, you are right